pkg/controller/common/reconciler.go

```diff
@@ -49,8 +71,25 @@ func (r *BaseIngressReconciler) ReconcileIngress(ctx context.Context, ci network
 	return nil
 }
 
-func (r *BaseIngressReconciler) deleteRoutes(ctx context.Context, ci networkingv1alpha1.IngressAccessor) error {
+func routeMap(routes routev1.RouteList) map[string]*routev1.Route {
+	mp := make(map[string]*routev1.Route, len(routes.Items))
+	for _, route := range routes.Items {
+		mp[route.Name] = &route
+	}
+	return mp
+}
+
+func (r *BaseIngressReconciler) deleteRoute(ctx context.Context, route *routev1.Route) error {
 	logger := logging.FromContext(ctx)
+	logger.Infof("Deleting OpenShift Route for host %s", route.Spec.Host)
+	if err := r.Client.Delete(ctx, route); err != nil {
+		return err
```

Comment:
Let's add the same context to the error as above here for consistency and easier debugability.